edb/pgsql/compiler/dml.py

```diff
@@ -1172,9 +1225,11 @@ def process_link_values(
         if iterator_cte is not None:
             iterator_rvar = relctx.rvar_for_rel(
                 iterator_cte, lateral=True, ctx=subrelctx)
-            relctx.include_rvar(row_query, iterator_rvar,
-                                path_id=iterator_cte.query.path_id,
-                                ctx=subrelctx)
+            # XXX???
+            if iterator_cte.query.path_id:
```

Comment:
Perhaps pass the `(iterator_cte, iterator_set)` tuple to avoid having to fetch the `path_id`?